test/e2e/parallel/proxy_test.go

```diff
@@ -1033,13 +1041,141 @@ func TestSpaceLister(t *testing.T) {
 		require.NoError(t, err)
 	})
 
-	t.Run("banned user cannot create proxy client", func(t *testing.T) {
-		// when
-		_, err := hostAwait.CreateAPIProxyClient(t, users["banneduser"].token, hostAwait.APIProxyURL)
+	t.Run("banned user", func(t *testing.T) {
+		bannedUser := users["banneduser"]
 
-		// then
-		// this is the error expected to be returned by the proxy when the user is banned
-		require.ErrorContains(t, err, "unable to get target cluster: no member cluster found for the user")
+		t.Run("cannot get apis list", func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s%s", hostAwait.APIProxyURL, "/apis"), nil)
+			require.NoError(t, err)
+
+			req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", bannedUser.token))
+			tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} // nolint:gosec
+			client := &http.Client{Transport: tr}
+			res, err := client.Do(req)
+			require.NoError(t, err)
+			defer res.Body.Close()
+
+			require.Equal(t, http.StatusForbidden, res.StatusCode)
+		})
+
+		t.Run("cannot get workspace", func(t *testing.T) {
+			_, err := bannedUser.getWorkspace(t, hostAwait, bannedUser.compliantUsername)
+
+			require.Error(t, err)
+			require.True(t, meta.IsNoMatchError(err), "expected NoMatch error, got %v", err)
+		})
+
+		t.Run("cannot list workspaces", func(t *testing.T) {
+			proxyCl := bannedUser.createProxyClient(t, hostAwait)
+
+			workspaces := &toolchainv1alpha1.WorkspaceList{}
+			err := proxyCl.List(context.TODO(), workspaces)
+
+			require.Error(t, err)
+			require.True(t, meta.IsNoMatchError(err), "expected NoMatch error, got %v", err)
+		})
+
+		t.Run("cannot list resources", func(t *testing.T) {
+			tt := map[string]string{
+				"home workspace":   "banneduser",
+				"shared workspace": "car",
+			}
+			for k, w := range tt {
+				t.Run(k, func(t *testing.T) {
+					proxyCl, err := hostAwait.CreateAPIProxyClient(t, bannedUser.token, hostAwait.ProxyURLWithWorkspaceContext(w))
+					require.NoError(t, err)
+
+					tt := map[string]client.ObjectList{
+						"applications": &appstudiov1.ApplicationList{},
+						"pods":         &corev1.PodList{},
+						"configmaps":   &corev1.ConfigMapList{},
+					}
+					for k, o := range tt {
+						t.Run(k, func(t *testing.T) {
+							err := proxyCl.List(context.TODO(), o)
+							require.Error(t, err)
+							require.True(t, meta.IsNoMatchError(err), "expected NoMatch error, got %v", err)
+						})
+					}
+				})
+			}
+		})
+	})
+
+	t.Run("lazy banned user", func(t *testing.T) {
```

Comment:
What is the difference comparing to `banneduser` tests?

Reply:
the difference is that `banneduser`'s client does not initialize correctly -successful call to `/apis`, so we expect it to return `NoMatch` errors. In this case the client initialized correctly and we have `Forbidden` errors.

Reply:
should we rename it to something like "user creates client and after is banned" ? But more important does it matter when the user is banned , if before or after the initialization of a kube client ? 

I'm suggesting that probably we don't need this test. WDYT?

Reply:
AFAIK, for how Controller-Runtime's Client v0.13.0 is implemented, the Controller-Runtime's client performs a call to the `/apis` endpoint during initialization. If we ban the user before creating the client, this call will fail and subsequent requests will fail with `NoMatch` without really performing the request against the proxy, as the client was not able to initialize the list of resources the cluster is supporting. 

Indeed, the [client.New()](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/client#New) function [instantiates a new Dynamic RESTMapper](https://github.com/kubernetes-sigs/controller-runtime/blob/d242fe21e646f034995c4c93e9bba388a0fdaab9/pkg/client/client.go#L108) (if not provided via client's Config).
The [mapper invokes](https://github.com/kubernetes-sigs/controller-runtime/blob/d242fe21e646f034995c4c93e9bba388a0fdaab9/pkg/client/apiutil/dynamicrestmapper.go#L85) the `/apis` endpoint to fetch the list of resources that the cluster is supporting. That's done [invoking the setStaticMapper function here](https://github.com/kubernetes-sigs/controller-runtime/blob/d242fe21e646f034995c4c93e9bba388a0fdaab9/pkg/client/apiutil/dynamicrestmapper.go#L99).
Next usages of the client will lookup the cached list of supported resources to understand if the cluster supports the provided resource and return a `NoMatch` error without really executing the request.

Conversely, is we ban the user lately, the client will effectively perform the call and the proxy will reject it with `Forbidden`.

As the former scenario is mainly testing a Controller-Runtime's Client specific behavior, I'd say we could remove the `banneduser` suite, and rename the `lazy banned user` as `banneduser`. WDYT?

Reply:
Thanks for the detailed explanation.
I don't think we need to test the client (initialization). We want to test our proxy here. So I agree with @mfrancisc and would drop the tests which fails with the client initialization.
Create the client. Then ban the user. Then call the proxy and check the expected 403 error.

Reply:
done in 89a664f52fc50c2ac8187c417e84bbb73384bfed